src/interface/contourplot.jl

```diff
@@ -0,0 +1,72 @@
+"""
+    contourplot(x, y, z; kwargs...)
+
+Description
+============
+
+Draws a contour plot on a new canvas. `levels` controls the number of contour levels.
+
+Usage
+=====
+    contourplot(x, y, z; kwargs...)
+
+Arguments
+=========
+
+- **`levels`** : The number of contour levels. 
+
+- **`colormap`** : The colormap to use for the contour lines. 
+
+- **`colorbar`** : Toggle the colorbar. 
+
+$DOC_PLOT_PARAMS
+
+- **`height`** : Number of character rows that should be used
+  for plotting.
+
+- **`xlim`** : Plotting range for the x axis.
+  `(0, 0)` stands for automatic.
+
+- **`ylim`** : Plotting range for the y axis.
+  `(0, 0)` stands for automatic.
+
+- **`grid`** : If `true`, draws grid-lines at the origin.
+
+Author(s)
+==========
+
+- T Bltg (https://github.com/t-bltg)
+"""
+
+function contourplot(
+    x::AbstractVector, y::AbstractVector, z::AbstractMatrix;
+    canvas::Type = BrailleCanvas, name::AbstractString = "", levels::Integer = 5, colormap = :viridis,
+    blend = false, grid = false, colorbar = true, kw...
+)   
+    callback = colormap_callback(colormap)
+    new_plot = Plot(
+        collect(extrema(x)),
+        collect(extrema(y)), canvas;
+        blend = blend, grid = grid, colorbar = colorbar, colormap = callback,
+        kw...
+    )
+    contourplot!(new_plot, x, y, z; name = name, levels = levels, colormap = callback)
+end
+
+function contourplot!(
+    plot::Plot{<:Canvas}, x::AbstractVector, y::AbstractVector, z::AbstractMatrix;
+    name::AbstractString = "", levels::Integer = 5, colormap = :viridis, 
```

Comment:
Will `levels = 5` be a reasonable default here for irregular data points?

Reply:
All the other plotting packages (`PyPlot`, `Plots`) use a higher number of levels, which is unreadable in `UnicodePlots`.
`5` is good since it get's you `0` level for real valued data. We can go down to `3`, but this is subjective. What be be your preference ?

Reply:
I went down to 3 since our plots are really small by default.